src/main/java/nextstep/subway/line/Sections.java

```diff
@@ -1,79 +1,131 @@
 package nextstep.subway.line;
 
-import nextstep.subway.line.exception.AlreadyRegisteredStationException;
+import nextstep.subway.line.exception.DuplicateStationException;
 import nextstep.subway.line.exception.LastOneSectionException;
 import nextstep.subway.line.exception.NotDownStationException;
-import nextstep.subway.line.exception.NotSameNewUpStationAndExistingDownStationException;
 import nextstep.subway.station.Station;
 import nextstep.subway.station.Stations;
 
 import javax.persistence.CascadeType;
 import javax.persistence.Embeddable;
 import javax.persistence.FetchType;
 import javax.persistence.OneToMany;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.*;
+import java.util.stream.IntStream;
 
 @Embeddable
 public class Sections {
 
     @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, mappedBy = "line", orphanRemoval = true)
-    private List<Section> sections = new ArrayList<>();
+    private final List<Section> sections;
 
     public Sections() {
+        sections = new ArrayList<>();
     }
 
     private Sections(Section section) {
+        sections = new ArrayList<>();
+        section.changeToFirst();
         sections.add(section);
     }
 
+    private Sections(Section... section) {
+        sections = new ArrayList<>(Arrays.asList(section));
+        sections.get(0).changeToFirst();
+    }
+
     public static Sections of(Line line, Station upStation, Station downStation, Integer distance) {
-        return new Sections(createFirstSection(line, upStation, downStation, distance));
+        return new Sections(createSection(line, upStation, downStation, distance));
+    }
+
+    public static Sections of(Section... section) {
+        return new Sections(section);
     }
 
-    private static Section createFirstSection(Line line, Station upStation, Station downStation, Integer distance) {
+    private static Section createSection(Line line, Station upStation, Station downStation, Integer distance) {
         return Section.builder()
                 .line(line)
                 .upStation(upStation)
                 .downStation(downStation)
                 .distance(distance)
-                .isFirst(true)
                 .build();
     }
 
     public void add(Section section) {
-        Section lastSection = findLastSection();
-        if (lastSection.notSameDownStation(section.getUpStation())) {
-            throw new NotSameNewUpStationAndExistingDownStationException();
+        if (findEndSection().sameDownStationAndUpStationOf(section)) {
```

Comment:
```suggestion
        if (sectionAddable(section)) {
```

이런식으로 구간 정보 추가에 대한 책임을 분리해도 가독성이 향상될 수 있습니다!

Reply:
확실히 리뷰어님의 말씀대로 하면 가독성이 높아지겠네요!